ksql-engine/src/main/java/io/confluent/ksql/streams/StreamsFactories.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License; you may not use this file
+ * except in compliance with the License.  You may obtain a copy of the License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.streams;
+
+import io.confluent.ksql.util.KsqlConfig;
+
+public class StreamsFactories {
+  private final GroupedFactory groupedFactory;
+  private final JoinedFactory joinedFactory;
+  private final MaterializedFactory materializedFactory;
+
+  public static StreamsFactories create(final KsqlConfig ksqlConfig) {
+    return new StreamsFactories(
+        GroupedFactory.create(ksqlConfig),
+        JoinedFactory.create(ksqlConfig),
+        MaterializedFactory.create(ksqlConfig)
+    );
+  }
+
+  public StreamsFactories(
```

Comment:
nit: can this be private / package local?

How about some null checks on the params?